x-pack/plugins/security_solution/server/lib/detection_engine/routes/privileges/read_privileges_route.ts

```diff
@@ -40,6 +40,12 @@ export const readPrivilegesRoute = (
         const clusterPrivileges = await readPrivileges(clusterClient.callAsCurrentUser, index);
         const privileges = merge(clusterPrivileges, {
           is_authenticated: security?.authc.isAuthenticated(request) ?? false,
+          can_read_actions:
+            (
+              await security?.authz?.checkPrivilegesDynamicallyWithRequest(request)({
+                kibana: [security?.authz?.actions.ui.get('actions', 'show')],
+              })
+            )?.hasAllRequested ?? false,
```

Comment:
Sorry for the drive-by comment. I could be overlooking something so feel free to ignore:

This information should already be available in the UI via the capabilities object. You could alternatively use that instead of making another round-trip to Elasticsearch to calculate this information:

```ts
public start(core: CoreStart) {
   const canReadActions = core.application.capabilities.actions.show;
}
```

Reply:
This is great thanks for the eyes! I didn't know about this. Would this provide user / role level information per request? I'm assuming yes.

Reply:
> Would this provide user / role level information per request? I'm assuming yes.

The capabilities object doesn't tell you who the current user is, or what their roles are. This object does reflect the current user's privileges though. Is that what you're looking for?

Reply:
Yes okay perfect. Thanks for the eyes! I'll make this change.